packages/@aws-cdk/aws-iot-actions/lib/location-action.ts

```diff
@@ -0,0 +1,160 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CfnTopicRule } from '@aws-cdk/aws-iot';
+import { CfnTracker } from '@aws-cdk/aws-location';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Location Timestamp Unit Enum
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+ */
+export enum LocationTimestampUnit {
+  /**
+   * SECONDS
+   */
+  SECONDS='SECONDS',
+
+  /**
+   * MILLISECONDS
+   */
+  MILLISECONDS='MILLISECONDS',
+
+  /**
+   * MICROSECONDS
+   */
+  MICROSECONDS='MICROSECONDS',
+
+  /**
+   * NANOSECONDS
+   */
+  NANOSECONDS='NANOSECONDS',
+}
+
+/**
+ * Describes how to interpret an application-defined timestamp value from an MQTT message payload and the precision of that value
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+ */
+export interface LocationTimestamp {
+  /**
+     * The precision of the timestamp value that results from the expression described in `value`.
+     * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+     *
+     * @default MILLISECONDS
+     */
+  readonly unit?: LocationTimestampUnit
+
+  /**
+     * An expression that returns a long epoch time value
+     * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+     *
+     * @default '${timestamp()}'
+     */
+  readonly value?: string
```

Comment:
Can you help me understand how this is used? I had intended to suggest using `Duration` here, but if a timestamp is really the thing or something else, that's a bad suggestion on my end. Either way, can we make this more strongly typed than a string?

Reply:
We don't put a `Date` or `number` in here but an _expression_. This expression returns an epoch timestamp. See the example below. But in short: We can use the [IoT `timestamp()` function](https://docs.aws.amazon.com/iot/latest/developerguide/iot-sql-functions.html#iot-function-timestamp) here. Meaning we will have a vale `value: '${timestamp()}` which will be turned into the current epoch timestamp of when the message was processed.

Reply:
Ah, I see. So there isn't really a good way to type this more strongly.